Add support for wildcards in the map file #170

marc0l92 opened the pull request:
I'm using the function `glob.sync()` (that you already use to find the mapping file) to expand the wildcards in the path to the example inside the mapping files.

codekie commented on src/index.js:
```diff
@@ -324,22 +324,32 @@ function _handleExamplesByMapValidation(openapiSpec, mapExternalExamples, statis
                     schema,
                     example,
                     statistics,
                     filePathExample
```
Passing the original `filePathExample` here, will also result in the glob showing up in the error messages for invalid examples (instead of the full path to the example file), which isn't very helpful:

e.g.

```
        {
           "dataPath": ".versions[0]"
           "exampleFilePath": "test/data/v2/external-examples-invalid-*.json"
           "keyword": "required"
           "mapFilePath": "/tmp/openapi-examples-validator/test/data/v2/map-external-examples-glob-invalid1.json"
           "message": "should have required property 'links'"
           "params": {
           "type": "Validation"
         }
```

`exampleFilePath` in the error-message should contain the expanded path to the invalid example file (e.g. `test/data/v2/external-examples-invalid-missing-link.json` in this case)

codekie reviewed (changes requested):
Please also add at least one test case for invalid examples and make sure that then output for globbed paths is the same as with the non-globbed counterparts (the ordering of files can be disregarded).

marc0l92 commented:
> Please also add at least one test case for invalid examples and make sure that then output for globbed paths is the same as with the non-globbed counterparts (the ordering of files can be disregarded).

How do you suggest me to do this?

codekie commented:
I'd suggest, you create two mapping files, one with and one without globs, selecting the same files and use the same reference values to test against.

Btw. I really appreciate your efforts 👍. Thanks

marc0l92 commented:
I created the test you asked

codekie commented on test/specs/impl/v2/index.js:
```diff
@@ -186,6 +190,52 @@ describe('Main-module, for v2 should', () => {
+            const resultWithWildcards = await validateExamplesByMap(FILE_PATH__EXTERNAL_EXAMPLES_SCHEMA,
+                FILE_PATH__EXTERNAL_EXAMPLES_MAP_WITH_WILDCARDS);
+            const resultWithoutWildcards = await validateExamplesByMap(FILE_PATH__EXTERNAL_EXAMPLES_SCHEMA,
+                FILE_PATH__EXTERNAL_EXAMPLES_MAP_WITH_WILDCARDS);
```
Isn't this supposed to be `FILE_PATH__EXTERNAL_EXAMPLES_MAP_WITHOUT_WILDCARDS`?

codekie reviewed (changes requested):
Just in case you made a force-push, please don't as GitHub can't automatically detect the changes since the last review.

marc0l92 commented:
> Just in case you made a force-push, please don't as GitHub can't automatically detect the changes since the last review.

Ok, in my company we try to limit the number of commits using push force, but I will not do it here

codekie commented:
The branch will be squashed to a single commit anyways when the pull request is merged, so it does not matter how many commits in the initial branch were 👍

marc0l92 commented:
You can merge
